packages/features/bookings/lib/handleNewBooking/getBookingData.ts

```diff
@@ -68,7 +68,12 @@ export async function getBookingData<T extends z.ZodType>({
     name: responses.name,
     email: responses.email,
     guests: responses.guests ? responses.guests : [],
-    location: responses.location?.optionValue || responses.location?.value || "",
+    location:
+      typeof responses.location === "object"
+        ? responses.location?.optionValue || responses.location?.value || ""
+        : typeof responses.location === "string"
```

Comment:
This is still a work around and does not address the main issue. When you change the label, it updates the type of location from radioInput to address which is what throws the error. Please fix that instead.

![Screenshot 2024-07-12 125722](https://github.com/user-attachments/assets/45805078-cdd4-417a-9bee-a5caede13148)

Reply:
Yes this is expected to happen as per https://github.com/calcom/cal.com/blob/c7c5cdf133bca0393d00314227de33957fb1e594/packages/features/form-builder/FormBuilder.tsx#L245
Hence the z.union is made for location field similar to name field, Based on user input , the location can be either an object (radioinput) or a string. This was not accomodated in Booking Form

Reply:
Ah my bad, I missed that

Reply:
No issues! @Amit91848 👍